presto-main/src/main/java/com/facebook/presto/execution/buffer/OutputBuffers.java

```diff
@@ -38,6 +38,9 @@
 public final class OutputBuffers
 {
     public static final int BROADCAST_PARTITION_ID = 0;
+    private static final OutputBuffers DISCARDING_OUTPUT_BUFFERS = createInitialEmptyOutputBuffers(BufferType.DISCARDING)
+            .withBuffer(new OutputBufferId(0), 0)
```

Comment:
You probably already noted that `OutputBufferId` corresponds to taskId . So, you see, when a stage has only 1 task, it's OK to assume the taskId of the only task is 0 ;)

 (the `OutputBuffers` provide the mapping from receiver task ID to sender's output buffer id)

Reply:
Actually you are right. Since this is DISCARDING output buffer, there will be no tasks that are reading for it. So it has to be simply initialized as `DISCARDING_OUTPUT_BUFFERS = createInitialEmptyOutputBuffers(DISCARDING).withNoMoreBufferIds();`